src/coreclr/vm/interpexec.cpp

```diff
@@ -1200,6 +1200,136 @@ void InterpExecMethod(InterpreterFrame *pInterpreterFrame, InterpMethodContextFr
                     UNREACHABLE();
                     break;
                 }
+                case INTOP_NEWARR:
+                {
+                    int32_t length = LOCAL_VAR(ip[2], int32_t);
+                    if (length < 0)
+                        assert(0); // Interpreter-TODO: Invalid array length
+
+                    MethodTable* pMT = (MethodTable*)pMethod->pDataItems[ip[3]];
+                    TypeHandle elemTH = TypeHandle(pMT);
+                    TypeHandle arrTH = elemTH.MakeSZArray();
```

Comment:
I don't think we should be doing this at run time. We should have this type already computed, rather than pass the element type